test/acceptance/spec/20-details.js

```diff
@@ -113,4 +137,46 @@ describe('Details Page', () => {
       browser.getValue('#dob').should.equal('01/01/2010');
     });
   });
+
+  describe('When I select the "Edit search" link on the details page', () => {
+    const child = expectedRecords.child;
+    const name = child.originalName;
+
+    before(() => {
+      mockProxy.willReturnForLocalTests(3);
+      browser.search('', name.surname, name.givenName, '');
+      const linkToFirstRecordDetails = "a[href=\"/details/" + expectedRecords.systemNumber + "?surname=" +
+        name.surname + "&forenames=" + name.givenName + "&multipleResults=true\"]";
+      browser.click(linkToFirstRecordDetails);
+      browser.click('#editSearchLink');
+});
+
+    it('returns the search page', () => {
+      browser.shouldBeOnSearchPage();
+    });
+
+    it('has the correct form values', () => {
+      browser.waitForVisible('input[name="forenames"]', 5000);
+      browser.getValue('#system-number').should.equal('');
+      browser.getValue('#surname').should.equal(name.surname);
+      browser.getValue('#forenames').should.equal(name.givenName);
+      browser.getValue('#dob').should.equal('');
+    });
+  });
+
+  describe('When I select the "Back to search results link on the results page"', () => {
+    it('returned me to the results page', () => {
+      const child = expectedRecords.child;
+      const name = child.originalName;
+
+      mockProxy.willReturnForLocalTests(3);
+      browser.search('', name.surname, name.givenName, '');
+      const linkToFirstRecordDetails = "a[href=\"/details/" + expectedRecords.systemNumber + "?surname=" +
+        name.surname + "&forenames=" + name.givenName + "&multipleResults=true\"]";
+      browser.click(linkToFirstRecordDetails);
+      browser.click('#backToSearchResults');
+      browser.getUrl().should.contain("surname=" + name.surname);
+      browser.getUrl().should.contain('forenames=' + name.givenName);
```

Comment:
How about adding in: browser.shouldBeOnResultsPage();

Which isn't a very rigorous test but it's better than nothing and could be improved in the future.